### PR TITLE
Specify interface ANY when the platform socket.if_nametoindex is absent.

### LINKED ISSUE #344: Platform may not provide interface indexes, breaking IPv6 support.
On some platforms the `ifaddr` module may not include an interface index. The default value for the `index` attribute on these platforms is `None`. This breaks IPv6 support where the interface index must be included in the encoding of the IP address.

I noticed this while investigating PyPy3 CI failures. I suspect that the standard release builds of PyPy are made on a platform without `if_nametoindex` support.

A possible fix for this would be to use the interface index 0 (meaning any) when no interface index is available.

### PR DESCRIPTION
**Pull request by ibygrave**
Closes #344

### CODE REVIEW

File: zeroconf/__init__.py [hunk 1]
```diff
@@ -2099,9 +2099,13 @@ def get_all_addresses() -> List[str]:
...
     # TODO: What about multi-address interfaces?
-    return list(
-        set((addr.ip, iface.index) for iface in ifaddr.get_adapters() for addr in iface.ips if addr.is_IPv6)
-    )
+    all_addresses = set()
+    for iface in ifaddr.get_adapters():
+        for addr in iface.ips:
+            if addr.is_IPv6:
+                # Unknown interface index -> "any" interface.
+                all_addresses.add((addr.ip, iface.index or 0))
```
Comment (bdraco) [inline zeroconf/__init__.py:2107]: Maybe make `0` a constant to show intent?

Comment (bdraco) [inline zeroconf/__init__.py:2107]: Maybe make `0` a constant to show intent?

**Comment by jstasiak** [comment]
Let's go back to #344 and discuss if this is actually needed.

**Comment by ibygrave** [comment]
Retracting

**Pull request closed without merging**